staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -33,21 +34,26 @@ type CustomResourceDefinitionSpec struct {
 	// +optional
 	Validation *CustomResourceValidation `json:"validation,omitempty" protobuf:"bytes,5,opt,name=validation"`
 	// Versions is the list of all supported versions for this resource.
+	// If Version field is provided, this field is optional.
 	// Validation: All versions must use the same validation schema for now. i.e., top
 	// level Validation field is applied to all of these versions.
 	// Order: The order of these versions is used to determine the order in discovery API
 	// (preferred version first).
-	Versions []CustomResourceDefinitionVersion `protobuf:"bytes,6,rep,name=versions"`
+	Versions []CustomResourceDefinitionVersion `json:"versions,omitempty" protobuf:"bytes,6,rep,name=versions"`
 }
 
 type CustomResourceDefinitionVersion struct {
 	// Name is the version name, e.g. “v1”, “v2beta1”, etc.
-	Name string `protobuf:"bytes,1,opt,name=name"`
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 	// Served is a flag enabling/disabling this version from being served via REST APIs
-	Served bool `protobuf:"varint,2,opt,name=served"`
-	// Storage flags the release as a storage version. There can be only one version
-	// flagged as Storage.
-	Storage bool `protobuf:"varint,3,opt,name=storage"`
+	// Default is True.
+	// +optional
+	Served *bool `json:"served,omitempty" protobuf:"varint,2,opt,name=served"`
+	// Storage flags the version as a storage version. There must be exactly one
+	// flagged as storage version.
+	// Default is False.
+	// +optional
+	Storage *bool `json:"storage,omitempty" protobuf:"varint,3,opt,name=storage"`
```

Comment:
Why don't we use a `StorageVersion` field on the top-level? This boolean seems to be redundant in n-1 cases.

Reply:
That was the discussed design that @liggitt suggested and @deads2k and @lavalamp agreed to.

Reply:
I'm envisioning v1 only containing `versions`, not a top-level `version` field, and wanted to structure the v1beta1 object like that as much as possible